### ISSUE TITLE
[BUG] eig3 backward produces non-finite gradients for repeated eigenvalues

### ISSUE THREAD

**Issue by christophercrouzet** [christophercrouzet/warp#16]
## What

The `eig3` adjoint divides by eigenvalue gaps. Its near-zero regularization excludes gaps that are exactly zero, so matrices with repeated eigenvalues can produce `inf` or `nan` gradients.

## Reproduction

```python
import numpy as np
import warp as wp

@wp.kernel
def f(a: wp.array[wp.mat33f], y: wp.array[wp.float32]):
    q = wp.mat33f()
    d = wp.vec3f()
    wp.eig3(a[0], q, d)
    y[0] = q[0, 1]

a = wp.array([np.eye(3, dtype=np.float32)], dtype=wp.mat33f, requires_grad=True, device="cpu")
y = wp.zeros(1, dtype=wp.float32, requires_grad=True, device="cpu")
with wp.Tape() as tape:
    wp.launch(f, dim=1, inputs=[a], outputs=[y], device="cpu")
tape.backward(y)
print(a.grad.numpy())
```

Current main prints an all-`nan` 3×3 input gradient. Zero matrices, scaled identities, and matrices such as `diag(1,1,2)` exercise the same exact-gap case.

## Why / when it matters

Repeated eigenvalues occur in isotropic tensors and symmetric models. A single such value can make an otherwise finite backward pass non-finite.

## Environment

Reproduced against NVIDIA/warp `main` at `b1922eba85bd32af5e393c829c0b22e70ea7abea` (Warp 1.18.0.dev5) on Linux.
